pkg/common/cns-lib/volume/manager.go

```diff
@@ -315,16 +315,60 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 		}
 	}
 
-	// Get the taskInfo.
-	taskInfo, err := cns.GetTaskInfo(ctx, task)
-	if err != nil || taskInfo == nil {
-		log.Errorf("failed to get taskInfo for CreateVolume task with err: %v", err)
-		if err != nil {
-			faultType = ExtractFaultTypeFromErr(ctx, err)
-		} else {
-			faultType = csifault.CSITaskInfoEmptyFault
+	taskInfo, err := task.WaitForResult(ctx, nil)
+	if err != nil {
+		if taskInfo == nil {
+			if !cnsvsphere.IsManagedObjectNotFound(err, task.Reference()) {
+				log.Errorf("failed to get taskInfo for CreateVolume task for "+
+					"volume %s with err: %v. Cannot determine if volume was created.",
+					volNameFromInputSpec, err)
+				return nil, csifault.CSITaskInfoEmptyFault, err
```

Comment:
Can you explain how the volume would get created again after we return from here?